bftengine/tests/simpleStorage/FileStorage.cpp

```diff
@@ -202,9 +216,12 @@ void FileStorage::writeInTransaction(uint16_t objectId, char *data, uint32_t dat
     LOG_ERROR(logger_, "FileStorage::writeInTransaction " << WRONG_FLOW);
     throw runtime_error(WRONG_FLOW);
   }
-  char *buf = new char[dataLength];
-  memcpy(buf, data, dataLength);
-  transaction_->insert(pair<uint16_t, RequestInfo> (objectId, RequestInfo(buf, dataLength)));
+  std::unique_ptr<char> buf(new char[dataLength]);
```

Comment:
In general, anytime we are creating a unique_ptr to wrap a char*, we should probably just use a vector instead. We can still memcpy into the vector by using it's `data()` member function.

It may be worth it in the future to go through the code and make these changes. I also think most current existing raw char* can be converted to vectors and make the code shorter, clearer, and safer.

Reply:
I'm not sure that vector's overhead worth that.

Reply:
Read a bit regarding this on the internet. Should be considered to be done as a separate task.

Reply:
👍